### PR TITLE
feat(metrics): add memory usage for streaming operators

### PR DESCRIPTION
**Pull request by yuhao-su**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## Checklist For Contributors

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [ ] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: src/stream/src/executor/monitor/streaming_stats.rs [hunk 1]
```diff
@@ -116,6 +116,9 @@ pub struct StreamingMetrics {
     // Materialize
     pub materialize_cache_hit_count: GenericCounterVec<AtomicU64>,
     pub materialize_cache_total_count: GenericCounterVec<AtomicU64>,
+
+    // Memory
+    pub stream_memory_usage: GenericGaugeVec<AtomicI64>,
```
Comment (st1page) [inline src/stream/src/executor/monitor/streaming_stats.rs:121]: it has not been updated in the grafana in this PR?

**Review by st1page — COMMENTED** [review]
generally LGTM, and you can consider adding these
1. watermark information in each LRU caches
2. executor name in the metrics info(tag)

File: risedev.yml [hunk 2]
```diff
@@ -42,8 +42,8 @@ profile:
       # - use: compactor
 
       # If you want to enable metrics, uncomment those two lines.
-      # - use: prometheus
-      # - use: grafana
+      - use: prometheus
```
Comment (wcy-fdu) [inline risedev.yml:45]: Please comment these two lines.

File: src/stream/src/cache/managed_lru.rs [hunk 3]
```diff
@@ -13,63 +13,221 @@
...
+
+    pub fn contains<Q>(&self, k: &Q) -> bool
+    where
+        KeyRef<K>: Borrow<Q>,
+        Q: Hash + Eq + ?Sized,
+    {
+        self.inner.contains(k)
+    }
+
+    pub fn len(&self) -> usize {
```
Comment (wcy-fdu) [inline src/stream/src/cache/managed_lru.rs:177]: `len` is more like length or capacity, this interface is the number of cached
```suggestion
    pub fn number_of_cached(&self) -> usize {
```

File: src/stream/src/executor/hash_join.rs [hunk 4]
```diff
@@ -820,10 +820,6 @@ impl<K: HashKey, S: StateStore, const T: JoinTypePrimitive> HashJoinExecutor<K,
                             .join_cached_entries
                             .with_label_values(&[&actor_id_str, side])
                             .set(ht.entry_count() as i64);
-                        // self.metrics
```
Comment (wcy-fdu) [inline src/stream/src/executor/hash_join.rs:823]: nits: remove this.

File: src/stream/src/cache/managed_lru.rs (hunk 3 above)
Comment (yuhao-su) [reply src/stream/src/cache/managed_lru.rs:177]: Rust convention is to use `len` as the number of elements in a collection

**Review by wcy-fdu — APPROVED** [review]
LGTM

**Review by st1page — APPROVED** [review]

**Pull request merged**